website/docs/components/switch/switch-a11y.md

```diff
@@ -11,47 +11,38 @@ tabs: Design('switch'), A11y('switch-a11y'), API('switch-api'), Example('switch-
-| switch |                        | `button` | When you can, use a `<button>` element and include `aria-pressed` or `role="switch"` for your toggles.                                                                         |
-|        | `aria-checked="true"` | `button` | Indicates the switch is on. CSS attribute selectors (for example [ `aria-checked="true"` ]) are used to synchronize the visual states with the value of the `aria-checked` attribute.   |
-|        | `aria-checked="false"` | `button` | Indicates the switch is off. CSS attribute selectors (for example [ `aria-checked="false"` ]) are used to synchronize the visual states with the value of the `aria-checked` attribute. |
-|        | `aria-readonly` | `button` | Indicates that the element isn’t editable, but is otherwise operable.                                                                                                         |
+| Component      | Role     | Attribute              | Usage                                                                                                                                                                                   |
+| -------------- | -------- | ---------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Switch.Value` | `switch` | `aria-checked="true"`  | Indicates the switch is on. |
+| `Switch.Value` | `switch` | `aria-checked="false"` | Indicates the switch is off. |
+| `Switch.Addon` |          | `aria-hidden="true"`   | Removes the text labels that appear next to the switch from the accessible name of the switch. This helps to prevent redundant announcement of the state by screen readers.                                                                                          |
 
 ## Considerations for designers
 
-* Make sure to add a non-color visual cue, like on-screen text, to convey the state of the switch. This will ensure that the meaning is clear to all users, including those who may have difficulty distinguishing colors.
-* If changing the state of a switch causes an instant change, it may violate a guideline called [WCAG Success Criterion 3.2.2 On Input](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html). To avoid this, either make sure the change doesn't automatically cause a [change of context](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html#dfn-changes-of-context), or inform users about the behavior before they use the switch.
+- Make sure to add a non-color visual cue, like on-screen text, to convey the state of the switch. This will ensure that the meaning is clear to all users, including those who may have difficulty distinguishing colors.
```

Comment:
switch has the "icon" prop which very conveniently helps with this, maybe worth mentioning it and recommending turning the icon prop on?

Reply:
oh, it's not a prop, but switch.value's child, but otherwise the comment stays relevant

Reply:
Thank you! Seems legit. I'll think on it